tests/ha/single_pod_restart/test_data_server_pod_restart_api.py

```diff
@@ -0,0 +1,321 @@
+        LOGGER.info("Step 6: Successfully started data, server pod and cluster is online.")
+        self.restore_pod = False
+        LOGGER.info("Step 7: Verifying response of background chunk upload process")
+        self.extra_files.append(chunk_obj_path)
+        while True:
+            resp = upload_op.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+            if isinstance(resp, bool):
+                break
+        if resp is None:
+            assert_utils.assert_true(False, "Background process of chunk upload failed")
+        assert_utils.assert_true(resp, "Failure observed in chunk upload during server pod restart")
+        LOGGER.info("Step 7: Successfully performed chuck upload during server pod restart")
+        LOGGER.info("Step 8: Download object which was uploaded in healthy cluster and verify "
+                    "checksum")
+        resp = self.ha_obj.object_download_jclient(s3_data=self.s3_clean,
+                                                   bucket_name=self.bucket_name,
+                                                   object_name=self.object_name,
+                                                   obj_download_path=download_path)
+        LOGGER.info("Download object response: %s", resp)
+        assert_utils.assert_true(resp[0], resp[1])
+        dnld_chksm = self.ha_obj.cal_compare_checksum(file_list=[download_path], compare=False)[0]
+        assert_utils.assert_equal(upld_chksm_hlt, dnld_chksm,
+                                  f"Expected checksum: {upld_chksm_hlt},"
+                                  f"Actual checksum: {dnld_chksm}")
+        LOGGER.info("Step 8: Successfully downloaded object and verified checksum")
+        LOGGER.info("Step 9: Download object which was uploaded in background and verify checksum")
+        LOGGER.info("Calculating checksum of uploaded file %s", chunk_obj_path)
+        upload_checksum = self.ha_obj.cal_compare_checksum(file_list=[chunk_obj_path],
+                                                           compare=False)[0]
+        resp = self.ha_obj.object_download_jclient(s3_data=self.s3_clean,
+                                                   bucket_name=bucket_name,
+                                                   object_name=object_name,
+                                                   obj_download_path=download_path)
+        LOGGER.info("Download object response: %s", resp)
+        assert_utils.assert_true(resp[0], resp[1])
+        download_checksum = self.ha_obj.cal_compare_checksum(file_list=[download_path],
+                                                             compare=False)[0]
+        assert_utils.assert_equal(upload_checksum, download_checksum,
+                                  f"Expected checksum: {upload_checksum},"
+                                  f"Actual checksum: {download_checksum}")
+        LOGGER.info("Step 9: Successfully downloaded object and verified checksum")
```

Comment:
can you form a dict of bucket name and object name, and then traverse through it for download object?
like: {self.bucket_name: self. object_name, bucket_name: object_name}
This will save 10-15 lines.